app/uk/gov/hmrc/gform/eval/EvaluationResults.scala

```diff
@@ -178,6 +182,36 @@ case class EvaluationResults(
 
     loop(expr)
   }
+  private def evalDateString(
+    expr: Expr,
+    recData: RecData[SourceOrigin.OutOfDate]
+  ): ExpressionResult = {
+
+    def nonEmpty(stringResult: StringResult): ExpressionResult =
+      if (stringResult.value.trim.isEmpty) Empty else stringResult
+
+    def fromVariadicValue(variadicValue: VariadicValue): ExpressionResult =
+      variadicValue.fold[ExpressionResult](one => nonEmpty(StringResult(one.value)))(many =>
+        ExpressionResult.OptionResult(many.value.map(_.toInt)))
+
+    def loop(expr: Expr): ExpressionResult = expr match {
+      case ctx @ FormCtx(formComponentId) =>
+        val exprResult: ExpressionResult = get(ctx, recData, fromVariadicValue)
+        exprResult.fold[ExpressionResult](identity)(_ => exprResult)(_ => {
+          val year = get(formComponentId.toAtomicFormComponentId(Atom("year")), recData)
+          val month = get(formComponentId.toAtomicFormComponentId(Atom("month")), recData)
+          val day = get(formComponentId.toAtomicFormComponentId(Atom("day")), recData)
+          (year, month, day) match {
+            case (Some(VariadicValue.One(y)), Some(VariadicValue.One(m)), Some(VariadicValue.One(d))) =>
+              StringResult(y.toInt.formatted("%04d") + m.toInt.formatted("%02d") + d.toInt.formatted("%02d"))
```

Comment:
Why not to store it here as `DateResult`?

Reply:
Yes, had the same idea. Will do the change

Reply:
done